src/widget/colormenu.cpp

```diff
@@ -0,0 +1,47 @@
+#include "widget/colormenu.h"
+#include "util/color/color.h"
+
+ColorMenu::ColorMenu(QWidget *parent, PredefinedColorsRepresentation* pColorRepresentation)
+        : QMenu(parent) {
+    // If another title would be more appropriate in some context, setTitle
+    // can be called again after construction.
+    setTitle(tr("Set color"));
+    useColorSet(pColorRepresentation);
+}
+
+void ColorMenu::useColorSet(PredefinedColorsRepresentation* pColorRepresentation) {
+    clear();
+    for (const auto& pColor : Color::kPredefinedColorsSet.allColors) {
+        if (*pColor == *Color::kPredefinedColorsSet.noColor) {
+            continue;
+        }
+
+        QAction* pColorAction = new QAction(pColor->m_sDisplayName);
```

Comment:
QAction(const QString &text, QObject* parent);
Has no default parent before QT 5.7. 
https://github.com/qt/qtbase/commit/536da5eb6ab1b4b07c13ad9f4c0f69dba5ca1624
I think we should uses "this" here 
or even better a make_parented<QAction>(pColor->m_sDisplayName, this); to avoid plain "new"

Reply:
Done: 5182187

Reply:
Actually that does not work. The QList::append call below requires a copy assignment operator, which is deleted in parented_ptr. std::shared_ptr does not make sense because the QMenu::addAction below requires a raw QAction* anyway. So I think it's best to leave these as raw QAction* pointers. This class is very simple; it is easy to reason that they do not leak.